internal/pkg/agent/application/upgrade/artifact/download/snapshot/downloader.go

```diff
@@ -82,13 +88,19 @@ func snapshotConfig(config *artifact.Config, versionOverride string) (*artifact.
 	}, nil
 }
 
-func snapshotURI(versionOverride string, config *artifact.Config) (string, error) {
+func snapshotURI(versionOverride *agtversion.ParsedSemVer, config *artifact.Config) (string, error) {
+	// do we support upgrade without specifying a target version ?
```

Comment:
What is this question here for? Does this need to be answered before this is truely ready?